controllers/user.go

```diff
@@ -0,0 +1,123 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	"github.com/lifeiscontent/webservice/models"
+)
+
+type userController struct {
+	userIDPattern *regexp.Regexp
+}
+
+func (uc userController) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
@zigara when I first saw this syntax, I read it as some magic function without a name is implementing a ServeHTTP interface. Is this correct? or am I flailing about with my arms in the air?

Reply:
You're adding a method named ServeHTTP to the userController structure, but it happens in this case it does satisfy the Handler (https://golang.org/pkg/net/http/#Handler) interface, meaning you can use an instance of this structure for anything that accepts a Handler argument.

Reply:
@zigara ah, that makes much more sense now. I'm going to try to formalize this in my own words, please correct me if I'm wrong, or you have more input.

a struct is used to encapsulate data and formalize it into a schema of some sort.

you can then add members to a struct like what we see above to interact with the struct.

now, one question, if I have a userController defined, could I then call `userController.ServeHTTP` directly?

Reply:
Yes, that sounds accurate.

You would need an instance of it first, but yes;
`go
u := newUserController()
u.ServeHTTP(blah, blah)
`

Reply:
@zigara are there such things as static members? e.g.`newUserController.Blah` in go?

Reply:
Nope, you would probably just store those in the package as their own types/methods.